tests/ServiceLayer.Mesh.Tests/Functions/FileRetryFunctionTests.cs

```diff
@@ -0,0 +1,216 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NHS.MESH.Client.Contracts.Services;
+using ServiceLayer.Mesh.Data;
+using ServiceLayer.Mesh.Functions;
+using ServiceLayer.Mesh.Messaging;
+using ServiceLayer.Mesh.Models;
+using ServiceLayer.Mesh.Configuration;
+using Xunit;
+
+namespace ServiceLayer.Mesh.Tests.Functions;
+
+public class FileRetryFunctionTests
+{
+    private readonly Mock<ILogger<FileRetryFunction>> _loggerMock;
+    private readonly Mock<IMeshInboxService> _meshInboxServiceMock;
+    private readonly Mock<IFileExtractQueueClient> _fileExtractQueueClientMock;
+    private readonly Mock<IFileTransformQueueClient> _fileTransformQueueClientMock;
+    private readonly Mock<IFileRetryFunctionConfiguration> _configuration;
+    private readonly ServiceLayerDbContext _dbContext;
+    private readonly FileRetryFunction _function;
+
+    public FileRetryFunctionTests()
+    {
+        _loggerMock = new Mock<ILogger<FileRetryFunction>>();
+        _meshInboxServiceMock = new Mock<IMeshInboxService>();
+        _fileExtractQueueClientMock = new Mock<IFileExtractQueueClient>();
+        _fileTransformQueueClientMock = new Mock<IFileTransformQueueClient>();
+        _configuration = new Mock<IFileRetryFunctionConfiguration>();
+
+        var options = new DbContextOptionsBuilder<ServiceLayerDbContext>()
+            .UseInMemoryDatabase(databaseName: Guid.NewGuid().ToString())
+            .ConfigureWarnings(warnings =>
+                warnings.Ignore(Microsoft.EntityFrameworkCore.Diagnostics.InMemoryEventId.TransactionIgnoredWarning))
+            .Options;
+
+        _dbContext = new ServiceLayerDbContext(options);
+
+        _configuration.Setup(c => c.StaleHours).Returns("12");
+
+        _function = new FileRetryFunction(
+            _loggerMock.Object,
+            _meshInboxServiceMock.Object,
+            _dbContext,
+            _fileExtractQueueClientMock.Object,
+            _fileTransformQueueClientMock.Object,
+            _configuration.Object
+        );
+    }
+
+    [Fact]
+    public async Task Run_EnqueuesDiscoveredFileOlderThan12Hours()
```

Comment:
This test is very similar to Run_EnqueuesExtractingFileOlderThan12Hours - suggest using xUnit Theory to cover both scenarios.

More generally, most of the test scenarios should apply to a pair of statuses (Discovered & Extracting) / (Extracted & Transforming), so many of the tests could be data-driven in this way